core/src/main/java/hudson/model/AsyncPeriodicWork.java

```diff
@@ -112,6 +120,11 @@ public final void doRun() {
 
                 logger.log(Level.FINE, "Finished {0}. {1,number} ms",
                         new Object[]{name, stopTime - startTime});
+                thread = null;
+                if (queueIfAlreadyRunning() && pending.getAndSet(false)) {
```

Comment:
or
```suggestion
                if (pending.getAndSet(false)) {
```
should suffice

Reply:
So you want to make this new behaviour the default? I wanted to be conservative and keep the previous behaviour by default, but if you don't foresee any problem with it...